Add CHANGELOG.md

rsanchezsaez opened the pull request:
This PR adds a `CHANGELOG.md` file as per [Issue #458](https://github.com/ResearchKit/ResearchKit/issues/458).

It was auto-generated using the very nice [github_changelog_generator](https://github.com/skywinder/github-changelog-generator), and then manually edited to remove irrelevant *closed issues*.

These are the issues I removed (it would be great if we could mark these with either `duplicate`, `question`, `invalid` or `wontfix` labels here on *GitHub*, as that'll cause *github_changelog_generator* to automatically ignore them on subsequent runs):

- htmlReviewContent and makePDFWithCompletionHandler [\#461](https://github.com/ResearchKit/ResearchKit/issues/461)
- Version 1.2 Build Errors in Xcode 6.4 Base SDK iOS 8.4 [\#452](https://github.com/ResearchKit/ResearchKit/issues/452)
- problem installing ResearchKit via Cocoapods for level \> 1.0 [\#447](https://github.com/ResearchKit/ResearchKit/issues/447)
- restarting an already taken survey: ORKOrderedTask? [\#364](https://github.com/ResearchKit/ResearchKit/issues/364)
- walking the results after a TaskViewController \(Swift\) [\#363](https://github.com/ResearchKit/ResearchKit/issues/363)
- CustomRecorder in ORKTest not working as intended [\#340](https://github.com/ResearchKit/ResearchKit/issues/340)
- Disable cancel button? [\#328](https://github.com/ResearchKit/ResearchKit/issues/328)
- Task Progress - API Design [\#322](https://github.com/ResearchKit/ResearchKit/issues/322)
- Could not cast value of type 'ORKTaskResult' to 'ORKSpatialSpanMemoryResult' [\#320](https://github.com/ResearchKit/ResearchKit/issues/320)
- HealthKit heart rate algorithm [\#319](https://github.com/ResearchKit/ResearchKit/issues/319)
- What is the offset for the timestamp float in the acceleration and device motion JSON output files [\#309](https://github.com/ResearchKit/ResearchKit/issues/309)
- \[Swift\] Introduction example does not work. [\#303](https://github.com/ResearchKit/ResearchKit/issues/303)
- flipToPreviousPage removes steps and results [\#296](https://github.com/ResearchKit/ResearchKit/issues/296)
- Step result cleared on 'skip' action [\#295](https://github.com/ResearchKit/ResearchKit/issues/295)
- Bump versions to 1.1 [\#222](https://github.com/ResearchKit/ResearchKit/issues/222)
- Delete 'revert-198-tests-for-orkconsentdocument' branch [\#213](https://github.com/ResearchKit/ResearchKit/issues/213)
- Split ORKDataLoggerTests and ORKDataLoggerManagerTests to a new test target? [\#150](https://github.com/ResearchKit/ResearchKit/issues/150)
- ResearchKit on The Changelog! [\#84](https://github.com/ResearchKit/ResearchKit/issues/84)
- Broken ResearchKit logo image on developer.apple.com/opensource/ [\#66](https://github.com/ResearchKit/ResearchKit/issues/66)
- Best Practices - wrong link  [\#50](https://github.com/ResearchKit/ResearchKit/issues/50)
- Add version tags [\#42](https://github.com/ResearchKit/ResearchKit/issues/42)
- Why not swift? [\#25](https://github.com/ResearchKit/ResearchKit/issues/25)
- Website has placeholders for feed and favicon in \<head\> [\#9](https://github.com/ResearchKit/ResearchKit/issues/9)

Also, marking any *bug-related closed issues* with the `bug` label will improve layout of the *changelog file*, as those will be moved from *Closed issues* to *Fixed bugs* section under each release.

YuanZhu-apple commented:
I feels this auto generate one is hard to read. For example `Tower of Hanoi` is one of the new feature in 1.2 release, but it is not possible to dig it out from this PR.

I imagine the changelog would focus on the *new features* and *major API changes*.

A very rough draft (could be incomplete):
 
1.2 release: 
  - New Active tasks: Hanoi, PSAT, timedWalk
  - Charts Library


1.1 release:
  - ORKNavigableOrderedTask
  - New Active tasks: ReactionTime and Audiometry
  - Support vertical slider in answer format
  - Image capture step
  - iPad support

Maybe we can elaborate each item a little bit and add missing items.

rsanchezsaez commented:
How about keeping both?

I feel the auto-generated `CHANGELOG.md` it's still easier to parse than digging thought *GitHub*'s web interface (and it's quite trivial to maintain).

I can later write an additional `release-notes.md` file based on the draft you just posted.

YuanZhu-apple commented:
https://github.com/ResearchKit/ResearchKit/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Amerged+
The merged PR list is quite similar to a detailed changelog.md.
Also if we can label each merged PR with `bugfix`/`enhancement`/`new feature`/`1.1`/`1.2`, then the PR filtering will be more powerful and dynamic than a text file.

Would it be a nice idea to expose those commonly used PR filtering URL to help understand recent changes? 

Additionally I can edit the PR's name to better describe the changes we accepted.

rsanchezsaez commented:
Labeling the *PRs* is a great idea. It'd be great if we also labeled all *open/closed bugs* as well. It'll be a little work, but I think it's worth it. Specially so interested developers can filter out the `question`, `invalid` or `wontfix` ones.

If everything is labeled, I can see little use for the auto-generated text file.

I'll be writing the more succinct release notes soon.

rsanchezsaez commented:
I have replaced the `CHANGELOG.md` commit by a `RELEASE-NOTES.md` commit, written in the style you suggested.

I have added information about the new features taken from the relevant *pull requests*. Please let me know if some of the task descriptions are too detailed and you'd prefer a shorter, simpler, summaries.

Also let me know if we add notes about additional changes. Some details that are not currently mentioned and may be worth a note are, for example, the scale enhancements in *release 1.2*, such as the *Text Choice Support*.

YuanZhu-apple commented:
Thanks @rsanchezsaez, will get this **release note** reviewed.

YuanZhu-apple commented:
@rsanchezsaez Can you add some words about xcode version with each release?

Xcode 7 for 1.2 release. 
And Xcode 6.3 for 1.1 release.

rsanchezsaez commented:
Done.

umerkhan-apple commented on RELEASE-NOTES.md:
```diff
@@ -0,0 +1,94 @@
+  - **Result Predicates**: the predicate-building methods in `ORKResultPredicate` now use the new `ORKResultSelector` class for unequivocally identifying a *question step result* or a *form item result*.
+
+   This eliminates ambiguity when matching results with the same inner scope identifier. For example, a *form item result* can have the same identifier as a *question step result* or as another *form item result* in a different *form step*, and you can now match them separately.
+
```
Might want to indent this over, so that it falls under `Result Predicates`.

rsanchezsaez replied:
Good catch, fixed.

umerkhan-apple commented:
r=m3!

YuanZhu-apple commented:
Hi @rsanchezsaez, one more thing. 
Can you mark the contributors' name with the new features? It is a recognition of contributors.

rsanchezsaez commented:
Done.

@coxy1989: If you wish your full name instead of your *GitHub* nickname to appear in the release notes, let know.

@YuanZhu-apple, @umerkhan-apple: I have marked your improvements as contributed by *Apple Inc*.

YuanZhu-apple commented:
Looks really good!

DanKeen commented:
Minor comment - can you split up "iPhone landscape support" and "iPad support" so they are two line items?

Thanks so much for taking this on!

rsanchezsaez commented:
@DanKeen: Yup, done.

YuanZhu-apple commented on RELEASE-NOTES.md:
```diff
@@ -0,0 +1,128 @@
+
+    *Contributed by [Julien Therier](https://github.com/julientherier).*
+
+    The *PSAT task* provides support for both the *Paced Auditory Serial Addition Test (PASAT)* and the *Paced Visual Serial Addition Test (PVSAT)* versions.
```
From our internal edit:
Please replace `support for` with `adaptations of`.

YuanZhu-apple commented on RELEASE-NOTES.md:
```diff
@@ -0,0 +1,128 @@
+
+    The *[PASAT](https://en.wikipedia.org/wiki/Paced_Auditory_Serial_Addition_Test)* is a neuropsychological test used to assess capacity and rate of information processing and sustained and divided attention.
+
+    This task is documented in the scientific literature ([[Fos et al., 2000]](http://www.ncbi.nlm.nih.gov/pubmed/11125707), [[Nagels et al., 2005]](http://www.ncbi.nlm.nih.gov/pubmed/15823678)) as a measure of the [*Multiple Sclerosis Functional Score*](http://www.nationalmssociety.org/For-Professionals/Researchers/Resources-for-Researchers/Clinical-Study-Measures/Multiple-Sclerosis-Functional-C ...
```
From our internal edit:
Please replace `This task is` with `Both tests are`.

YuanZhu-apple commented on RELEASE-NOTES.md:
```diff
@@ -0,0 +1,128 @@
+
+    This task is documented in the scientific literature ([[Fos et al., 2000]](http://www.ncbi.nlm.nih.gov/pubmed/11125707), [[Nagels et al., 2005]](http://www.ncbi.nlm.nih.gov/pubmed/15823678)) as a measure of the [*Multiple Sclerosis Functional Score*](http://www.nationalmssociety.org/For-Professionals/Researchers/Resources-for-Researchers/Clinical-Study-Measures/Multiple-Sclerosis-Functional-C ...
+
+    It generates a series of single digits (for example, 60), at the specific frequency (for example, one new digit every 2 or 3 seconds). The user must add the newly presented digit to the one prior to it.
```
From our internal edit:
Please replace `It generates` with `This task generates`.

YuanZhu-apple commented on RELEASE-NOTES.md:
```diff
@@ -0,0 +1,128 @@
+
+    *Contributed by [Julien Therier](https://github.com/julientherier).*
+
+    This task measures gait speed, in the form of the [Timed 25-Foot Walk](http://www.nationalmssociety.org/For-Professionals/Researchers/Resources-for-Researchers/Clinical-Study-Measures/Timed-25-Foot-Walk-(T25-FW)) in the context of *multiple sclerosis*.
```
From our internal edit:
Please replace `in the form` with `and is an adaptation`.

YuanZhu-apple commented on RELEASE-NOTES.md:
```diff
@@ -0,0 +1,128 @@
+
+    *Contributed by [coxy1989](https://github.com/coxy1989).*
+
+    The [*Simple Reaction Time (SRT)*](http://www.cambridgecognition.com/tests/simple-reaction-time-srt) is a test which measures simple reaction time through delivery of a known stimulus to a known location to elicit a known response.
```
From our internal edit:
Please replace `The Simple Reaction Time (SRT) is a test` with `The Reaction Time Task is an adaptation of the Simple Reaction Time (SRT) test`.

YuanZhu-apple commented on RELEASE-NOTES.md:
```diff
@@ -0,0 +1,128 @@
+
+    *Contributed by [Vincent Tourraine](https://github.com/vtourraine).*
+
+    The [*Pure Tone Audiometry (PTA)*](https://en.wikipedia.org/wiki/Pure_tone_audiometry) test is a key hearing test used to identify hearing threshold levels of an individual, enabling determination of the degree, type and configuration of a hearing loss.
```
From our internal edit:
Please replace `The Pure Tone Audiometry (PTA) test` with `The Tone Audiometry Task is an adaptation of the Pure Tone Audiometry (PTA) test, which`.

chrisvariety commented on RELEASE-NOTES.md:
```diff
@@ -0,0 +1,128 @@
+
+ The views in the *Charts module* can be used independently of the rest of *ResearchKit*. It doesn't automatically connect with any other *ResearchKit* module: the developer has to supply the data to be displayed through the views' `dataSources`, which allows for maximum flexibility.
+
+- **Other improvemnets**
```
Typo here, should be`improvements`

YuanZhu-apple commented:
@rsanchezsaez Thanks! 
Here are final edits, after those changes applied, this can be merged.

rsanchezsaez commented:
I pushed a new commit taking care of the latest feedback. I also included some additional minor improvements (see commit message).

chrisvariety commented on RELEASE-NOTES.md:
```diff
@@ -0,0 +1,130 @@
+
+*ResearchKit 1.1* supports *iOS* and requires *Xcode 6.3* or newer. The minimum supported *Base SDK* is *8.0*.
+
+In addition to general stability and performance improvmenets, *ResearchKit 1.1* includes the following new features and enhancements.
```
Another typo here: `improvements`

rsanchezsaez commented:
@chrisvariety: Good catches! Fixed.

YuanZhu-apple commented:
Thanks @rsanchezsaez ! Merging it.

adamstac commented:
@rsanchezsaez thanks for pulling #84 into the details for this PR. We're looking forward to talking about ResearchKit on The Changelog!
